### ISSUE TITLE
Bug in PML-SilverMuller assert

### ISSUE THREAD

**Issue by lucafedeli88** [ECP-WarpX/WarpX#4269]
Hi, 
thanks to `clang-tidy` (https://github.com/ECP-WarpX/WarpX/pull/4268) I've found out this issue:

```
            WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
                !(
                    ( WarpX::field_boundary_lo[idim] == FieldBoundaryType::PML &&
                    WarpX::field_boundary_lo[idim] == FieldBoundaryType::Absorbing_SilverMueller ) ||
                    ( WarpX::field_boundary_hi[idim] == FieldBoundaryType::PML &&
                     WarpX::field_boundary_hi[idim] == FieldBoundaryType::Absorbing_SilverMueller )
                ),
                "PML and Silver-Mueller boundary conditions cannot be activated at the same time.");
```

The assert is always true because a boundary condition cannot be `PML` and `SilverMuller` at the same time. What was the intent of the check?

**Comment by dpgrote** [comment]
Perhaps this should be
```
WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
                !(
                    ( WarpX::field_boundary_lo[idim] == FieldBoundaryType::PML &&
                    WarpX::field_boundary_hi[idim] == FieldBoundaryType::Absorbing_SilverMueller ) ||
                    ( WarpX::field_boundary_hi[idim] == FieldBoundaryType::PML &&
                     WarpX::field_boundary_lo[idim] == FieldBoundaryType::Absorbing_SilverMueller )
                ),
                "PML and Silver-Mueller boundary conditions cannot be activated at the same time.");
```
Making sure that one end isn't one type and the other end the other.

**Comment by RemiLehe** [comment]
When tracking this, it seems that this was added here:
https://github.com/ECP-WarpX/WarpX/pull/2051/files#diff-1450fa5afa66d2e4dde0bd48fe21eb084d339554ad937d3604376713e053e3c4

**Comment by RemiLehe** [comment]
It seems that the original intent was: if **any** boundary is PML **and** **any** other boundary is Silver-Mueller, then abort.

**Comment by lucafedeli88** [comment]
This PR should fix the issue: https://github.com/ECP-WarpX/WarpX/pull/4297

**Comment by EZoni** [comment]
Fixed by #4297.
